esp/esp/utils/auth_backend.py

```diff
@@ -0,0 +1,25 @@
+from django.contrib.auth.backends import ModelBackend
+
+from esp.users.models import ESPUser
+
+class ESPAuthBackend(ModelBackend):
+    """ Subclass of Django's ModelBackend that returns an ESPUser
+    instead of a User. Adapted from
+    http://stackoverflow.com/questions/10682414/django-user-proxy-model-from-request
+    and from django/contrib/auth/backends.py as of 1.8.5 """
```

Comment:
Might want a comment somewhere explaining why we need both this and the custom middleware.  (I'm assuming it's so that this part can get the custom user, and that one can get the custom anonymous user?)